Source/WebCore/platform/graphics/FontMetrics.h

```diff
@@ -27,153 +27,146 @@ namespace WebCore {
-    
-    int capHeight() const { return m_intCapHeight; }
-    
-    bool hasIdenticalAscentDescentAndLineGap(const FontMetrics& other) const
-    {
-        return ascent() == other.ascent() && descent() == other.descent() && lineGap() == other.lineGap();
-    }
 
     std::optional<float> zeroWidth() const { return m_zeroWidth; }
     void setZeroWidth(float zeroWidth) { m_zeroWidth = zeroWidth; }
 
-    float ideogramWidth() const { return m_ideogramWidth; }
+    std::optional<float> ideogramWidth() const { return m_ideogramWidth; }
     void setIdeogramWidth(float ideogramWidth) { m_ideogramWidth = ideogramWidth; }
 
-    float underlinePosition() const { return m_underlinePosition; }
+    std::optional<float> underlinePosition() const { return m_underlinePosition; }
     void setUnderlinePosition(float underlinePosition) { m_underlinePosition = underlinePosition; }
 
-    float underlineThickness() const { return m_underlineThickness; }
+    std::optional<float> underlineThickness() const { return m_underlineThickness; }
     void setUnderlineThickness(float underlineThickness) { m_underlineThickness = underlineThickness; }
 
+    bool hasIdenticalAscentDescentAndLineGap(const FontMetrics& other) const
+    {
+        return intAscent() == other.intAscent()
+            && intDescent() == other.intDescent()
+            && intLineGap() == other.intLineGap();
+    }
+
 private:
     friend class Font;
 
     void reset()
     {
         m_unitsPerEm = defaultUnitsPerEm;
-        m_floatAscent = 0;
-        m_floatDescent = 0;
-        m_floatLineGap = 0;
-        m_floatLineSpacing = 0;
-        m_floatCapHeight = 0;
+
+        m_ascent = std::nullopt;
+        m_capHeight = std::nullopt;
+        m_descent = std::nullopt;
+        m_ideogramWidth = std::nullopt;
+        m_lineGap = std::nullopt;
+        m_lineSpacing = std::nullopt;
+        m_underlinePosition = std::nullopt;
+        m_underlineThickness = std::nullopt;
+        m_xHeight = std::nullopt;
+        m_zeroWidth = std::nullopt;
+
         m_intAscent = 0;
         m_intDescent = 0;
         m_intLineGap = 0;
         m_intLineSpacing = 0;
         m_intCapHeight = 0;
-        m_xHeight = 0;
-        m_zeroWidth = std::nullopt;
-        m_ideogramWidth = 0;
-        m_underlinePosition = 0;
-        m_underlineThickness = 0;
     }
 
     unsigned m_unitsPerEm { defaultUnitsPerEm };
 
-    float m_floatAscent { 0 };
-    float m_floatDescent { 0 };
-    float m_floatLineGap { 0 };
-    float m_floatLineSpacing { 0 };
-    float m_floatCapHeight { 0 };
+    std::optional<float> m_ascent;
+    std::optional<float> m_capHeight;
+    std::optional<float> m_descent;
+    std::optional<float> m_ideogramWidth;
+    std::optional<float> m_lineGap;
+    std::optional<float> m_lineSpacing;
+    std::optional<float> m_underlinePosition;
+    std::optional<float> m_underlineThickness;
+    std::optional<float> m_xHeight;
+    std::optional<float> m_zeroWidth;
```

Comment:
Using `std::optional<float>` will use way more memory than plain `float`. We should use `Markable<float>` to keep the same memory footprint.